client/gutenberg/extensions/wpcom-color-scheme/index.jsx

```diff
@@ -0,0 +1,93 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { registerBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import getColorSchemesData from 'blocks/color-scheme-picker/constants';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormRadiosBar from 'components/forms/form-radios-bar';
+import 'blocks/color-scheme-picker/style.scss';
+
+class WpcomColorScheme extends Component {
+	state = {
+		colorSchemePreference: null,
+	};
+
+	componentDidMount() {
+		apiFetch( {
+			path: '/me/settings?http_envelope=1',
+		} ).then( response => {
+			this.setState( {
+				colorSchemePreference: response.calypso_preferences.colorScheme,
+			} );
+		} );
+	}
+
+	updateColorScheme = event => {
+		this.setState( {
+			colorSchemePreference: event.target.value,
+		} );
+
+		const data = {
+			calypso_preferences: {
+				colorScheme: event.target.value,
+			},
+		};
+		// This currently requires different treatment for Calypso and wp-admin
+		const requestBody =
+			typeof window.wp !== 'undefined'
+				? {
+						headers: {
+							'Content-Type': 'application/json',
+						},
+						body: JSON.stringify( data ),
+				  }
+				: {
+						body: data,
+				  };
```

Comment:
why was this `headers` thing necessary? is something broken elsewhere in `wpcom-proxy-request`

Reply:
@dmsnell might have a point tho -- should we fix this in a more centralized/upstream place?

Reply:
Sure thing - this PR is nowhere near merging, it's only a PoC. Fixing that in the middleware would be ideal.